src/middlewared/middlewared/plugins/network.py

```diff
@@ -1925,13 +1925,13 @@ async def sync(self, wait_dhcp=False):
             await self.middleware.call('interface.pre_bridge_setup', bridge, sync_interface_opts)
 
         # Set VLAN interfaces MTU last as they are restricted by underlying interfaces MTU
-        for interface in sorted(
-            filter(lambda i: not i.startswith('br'), interfaces), key=lambda x: x.startswith('vlan')
-        ):
-            try:
-                await self.sync_interface(interface, wait_dhcp, sync_interface_opts[interface])
-            except Exception:
-                self.logger.error('Failed to configure {}'.format(interface), exc_info=True)
+        vlans, non_vlans = bisect(lambda x: x.startswith('vlan'), filter(lambda i: not i.startswith('br'), interfaces))
+        if non_vlans:
+            await asyncio.wait([self._sync_interface_wrapper(interface, wait_dhcp, sync_interface_opts[interface])
+                                for interface in non_vlans])
+        if vlans:
+            await asyncio.wait([self._sync_interface_wrapper(interface, wait_dhcp, sync_interface_opts[interface])
+                                for interface in vlans])
```

Comment:
Wont we remain with the same problem here? waiting dhcp on an unconfigured interface before we set up bridges?

Reply:
@william-gr all this code is executed only for configured interfaces. The interface with no carrier was explicitly configured for DHCP.

Reply:
Same question applies

Reply:
If a DHCP times out for an interface that is not related to bridge, bridge will only be configured after timeout of that unrelated interface, no?